python/ray/data/tests/test_dataset.py

```diff
@@ -1873,6 +1916,62 @@ def test_map_batch(ray_start_regular_shared, tmp_path):
             lambda df: 1, batch_size=2, batch_format="pyarrow"
         ).take()
 
+    # Test extra UDF args.
+    df = pd.DataFrame({"one": [1, 2, 3], "two": [2, 3, 4]})
+    table = pa.Table.from_pandas(df)
+    pq.write_table(table, os.path.join(tmp_path, "test1.parquet"))
+
+    def udf(batch, a, b=None):
+        assert a == 1
+        assert b == 2
+        return b * batch + a
+
+    ds = ray.data.read_parquet(str(tmp_path))
+    ds2 = ds.map_batches(
+        udf,
+        batch_size=1,
+        batch_format="pandas",
+        fn_args=(ray.put(1),),
```

Comment:
Add cases for passing args via just one of them?

Reply:
Hmm doesn't the args propagating successfully from both imply that the args will successfully propagate for each individually? I can't see how a failure scenario could arise where passing both succeeds but passing e.g. only `fn_args` would fail.

Reply:
Maybe, the issue is we never know and things can happen in unexpected ways. If we take an agnostic view about internals and just look at the new map_batches() API, and try to design test cases for it, I'd think this would be an angle to cover.

Reply:
Tests are geared towards the semantics of the API, and the semantics are that these are independent passthrough variables, so it's hard to see how the implementation could deviate to the point where a single test asserting that both are passed through would pass while the individual cases would fail. There's a development and maintenance cost to each test written, so I was trying to be pragmatic here.

Anyway, I'll add individual arg test coverage to resolve this.